teku/src/main/java/tech/pegasys/teku/cli/deposit/DepositGenerateAndRegisterCommand.java

```diff
@@ -96,8 +97,34 @@ public void run() {
   @NotNull
   private Function<ValidatorKeys, SafeFuture<TransactionReceipt>> registerValidator(
       final RegisterAction registerAction) {
-    return validatorKey ->
-        registerAction.sendDeposit(
-            validatorKey.getValidatorKey(), validatorKey.getWithdrawalKey().getPublicKey());
+    return validatorKey -> {
+      final BLSKeyPair validatorKeyPair = validatorKey.getValidatorKey();
+      final BLSKeyPair withdrawalKeyPair = validatorKey.getWithdrawalKey();
+      if (displayConfirmation) {
+        SUB_COMMAND_LOG.display(
+            String.format(
+                "%nSending deposit for Validator Key [%s].%n",
+                validatorKeyPair.getPublicKey().toString()));
+      }
+
+      final SafeFuture<TransactionReceipt> transactionReceiptSafeFuture =
+          registerAction.sendDeposit(validatorKeyPair, withdrawalKeyPair.getPublicKey());
+
+      if (displayConfirmation) {
+        transactionReceiptSafeFuture.finish(
+            transactionReceipt ->
+                SUB_COMMAND_LOG.display(
+                    String.format(
+                        "Transaction for Validator Key [%s] Completed. Transaction Hash: [%s]%n",
+                        validatorKeyPair.getPublicKey().toString(),
+                        transactionReceipt.getTransactionHash())),
+            exception ->
+                SUB_COMMAND_LOG.error(
+                    String.format(
+                        "Transaction for Validator Key [%s] Failed: Message: [%s]%n",
+                        validatorKeyPair.getPublicKey().toString(), exception.getMessage())));
+      }
+      return transactionReceiptSafeFuture;
```

Comment:
This I think could be part of the sendDeposit so it could be reused between both commands

Reply:
refactored down the stack.